### PR TITLE
Added recipe for ggplot

### PR DESCRIPTION
**Pull request by pmlandwehr**
Despite the recipe in the commit log, this recipe *is* working. I'm not clear why `conda build` sometimes fails on this, omitting all build steps, but it does. Anyhoo, point is: this is a good recipe for the python port of ggplot.

### CODE REVIEW

**Comment by pmlandwehr** [comment]
The build breaks right now because there's no recipe for brewer2mpl - should work once #804 is resolved.

**Comment by jakirkham** [comment]
Could we please rebase this? All the merge commits are a bit messy and am not really sure what value they provide.

File: recipes/ggplot/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record=record.txt
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - brewer2mpl
+    - cycler
```
Comment (ocefpaf) [inline recipes/ggplot/meta.yaml]: `matplotlib` should install `cycler`, no?

File: recipes/ggplot/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    - matplotlib
+    - numpy
+    - pandas
+    - patsy >=0.4
+    - scipy
+    - six
```
Comment (ocefpaf) [inline recipes/ggplot/meta.yaml]: You will probably need `six` at run time too.

File: recipes/ggplot/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  script: python setup.py install --single-version-externally-managed --record=record.txt
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - brewer2mpl
+    - cycler
+    - matplotlib
+    - numpy
```
Comment (ocefpaf) [inline recipes/ggplot/meta.yaml]: `matplotlib` also install `numpy`

File: recipes/ggplot/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record=record.txt
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - brewer2mpl
```
Comment (ocefpaf) [inline recipes/ggplot/meta.yaml]: When using `--single-version-externally-managed` you probably don't need to list the run time dependencies at build time (unless the package imports itself). Try removing all but `python` and `setuptools` and let's see if that works here.

File: recipes/ggplot/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    - ggplot
+
+about:
+  home: https://github.com/yhat/ggplot
+  license: BSD 2-Clause
+  summary: 'ggplot for python'
+
+extra:
+  recipe-maintainers:
+    - pmlandwehr
```
Comment (ocefpaf) [inline recipes/ggplot/meta.yaml]: You can add me here too.

File: recipes/ggplot/meta.yaml (hunk 1 above)
Comment (pmlandwehr) [inline recipes/ggplot/meta.yaml]: Did some rebasing to my master branch, @jakirkham but frankly I'd prefer someone else handle it with this commit.

**Comment by ocefpaf** [comment]
`ggplot` imports itself to get the version :unamused: 
You will need to duplicate all the run time dependencies at build time. (Sorry.)
```
Traceback (most recent call last):
  File "setup.py", line 3, in <module>
    from ggplot.__init__ import __version__
  File "/opt/conda/conda-bld/work/ggplot-0.9.4/ggplot/__init__.py", line 16, in <module>
    _set_mpl_backend()
  File "/opt/conda/conda-bld/work/ggplot-0.9.4/ggplot/__init__.py", line 9, in _set_mpl_backend
    import matplotlib as mpl
ImportError: No module named matplotlib
Command failed: /bin/bash -x -e /opt/conda/conda-bld/work/ggplot-0.9.4/conda_build.sh
```

**Comment by ocefpaf** [comment]
Thanks @pmlandwehr!

**Pull request merged**